### ISSUE TITLE
question: why not spread instead of  reflect.apply?

### ISSUE THREAD

**Issue by jimmywarting** [sindresorhus/eslint-plugin-unicorn#1739]
### Description

I was reading upon this proposal: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/HEAD/docs/rules/prefer-reflect-apply.md#prefer-reflectapply-over-functionapply

why not suggest using something like `foo(...arr)`?

### Fail

...

### Pass

...

**Comment by sindresorhus** [comment]
In your example, spreading makes more sense, but spreading does not allow decided the `this` context.

**Comment by sindresorhus** [comment]
We could potentially add support for `.apply` in https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md

**Comment by jimmywarting** [comment]
hmm, you are right about the context thingy... wondering if there is cases where we can detect if the context is not necessary

for instances if the context (`console`) is the same as the one coming before the `log` function is the same, then...
```js
console.log.apply(console, arguments) // This could for sure be sorten to 
console.log(...arguments) // to this
```


----



Bit of topic, but i also wonder how different rules can take priority over one and the other... i mean prefer spread and prefer reflect is touching the same issue basically.
 
eslint should have something like a number prefix where you can decide that a number has less priority over another number or something...
```json
{
  "100-unicorn/prefer-reflect": [],
  "50-unicorn/prefer-spread": []
}
```

**Comment by sindresorhus** [comment]
I think we should leave this to ESLint's built-in `prefer-spread` rule. It already handles the safe `.apply()` cases where spread preserves the `this` value, like `foo.apply(null, args)` and `object.method.apply(object, args)`, and intentionally skips cases where the `this` binding would change.

Duplicating that logic in Unicorn would create overlap between ESLint core `prefer-spread`, `unicorn/prefer-spread`, and `unicorn/prefer-reflect-apply`. I added a note to the `prefer-reflect-apply` docs pointing users to ESLint's `prefer-spread` for this behavior.
